src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp

```diff
@@ -129,14 +100,137 @@ class UpdateCVEDescription final
             {
                 flatbuffers::FlatBufferBuilder builder;
 
-                auto vulnerabilityDescriptionFB =
-                    NSVulnerabilityScanner::CreateVulnerabilityDescriptionDirect(builder,
-                                                                                 VulnDescFBScoreBase,
-                                                                                 VulnDescFBClassificationStr.c_str(),
-                                                                                 VulnDescFBDescriptionStr.c_str(),
-                                                                                 VulnDescFBSeverityStr.c_str(),
-                                                                                 VulnDescFBScoreVersionStr.c_str(),
-                                                                                 VulnDescFBReferenceStr.c_str());
+                if (metricsArray)
+                {
+                    for (const auto& field : *metricsArray)
+                    {
+                        auto metricCVSSV3_1 = field->cvssV3_1();
+                        auto metricCVSSV3_0 = field->cvssV3_0();
+                        auto metricCVSSV2_0 = field->cvssV2_0();
+                        if (metricCVSSV3_1)
+                        {
+                            VulnDescFBScoreBase = metricCVSSV3_1->baseScore();
+                            auto baseSeverity = metricCVSSV3_1->baseSeverity();
+                            VulnDescFBSeverityStr = (nullptr != baseSeverity) ? baseSeverity->str() : "";
```

Comment:
We can avoid the "baseSeverity" variable declaration and also the nullptr comparison to make the code cleaner.
Example of the replacement of the 2 previous lines:
`VulnDescFBSeverityStr = metricCVSSV3_1->baseSeverity()? metricCVSSV3_1->baseSeverity()->str() : "";`

These changes could be applied to all similar cases.

Reply:
I'm no pretty sure if it is entirely correct to treat a null as a boolean, but I did the change.

Reply:
Yes, it is valid to convert a pointer value to a boolean. Here is an extract of the C++ 11 standard:
![image](https://github.com/wazuh/wazuh/assets/101227434/dc370521-4586-49c6-ab46-6e35c4a83ca2)